src/data_request_spec.rs

```diff
@@ -1,49 +1,155 @@
 use anyhow::{Context, Result};
+
+use polars::{
+    frame::DataFrame,
+    lazy::{
+        dsl::{col, lit},
+        frame::IntoLazy,
+    },
+    prelude::NamedFrom,
+    series::Series,
+};
 use serde::{Deserialize, Serialize};
 use std::{
     ops::{Index, IndexMut},
     str::FromStr,
 };
 
-use crate::{metadata::SourceDataRelease, parquet::MetricRequest};
+use crate::{
+    metadata::{Metadata, MetricId},
+    parquet::MetricRequest,
+};
 
 #[derive(Serialize, Deserialize, Debug)]
 pub struct DataRequestSpec {
+    pub geometry: GeometrySpec,
     pub region: Vec<RegionSpec>,
     pub metrics: Vec<MetricSpec>,
+    pub years: Option<Vec<String>>,
+}
+
+/// This is the respone to requesting a set of metrics
+/// it contains the matched Metric Requests along with
+/// the geometry that was selected we can also potentially
+/// use this to give feedback on other options in the future
+pub struct MetricRequestResult {
+    pub metrics: Vec<MetricRequest>,
+    pub selected_geometry: String,
 }
 
 impl DataRequestSpec {
-    pub fn metric_requests(&self, catalogue: &SourceDataRelease) -> Result<Vec<MetricRequest>> {
-        let mut metric_requests: Vec<MetricRequest> = vec![];
-        println!("Try to get metrics {:#?}", self.metrics);
+    /// Generates a vector of metric requests from a `DataRequestSpec`
+    /// and a catalouge.
```

Comment:
Suggest single line and typo
```suggestion
    /// Generates a vector of metric requests from a `DataRequestSpec` and a catalog.
```